timeutil/schedule.go

```diff
@@ -233,3 +475,242 @@ func ParseLegacySchedule(scheduleSpec string) ([]*Schedule, error) {
+	var parsed ClockSpan
+	var rest string
+
+	// timespan = time ( "-" / "~" ) time [ "/" ( time / count ) ]
+
+	parsed.Split, rest, err = parseCount(s)
+	if err != nil {
+		return span, fmt.Errorf("cannot parse %q: not a valid interval", s)
+	}
+
+	if strings.Contains(rest, spreadToken) {
+		// timespan uses "~" to indicate that the actual event
+		// time is to be spread.
+		parsed.Spread = true
+		rest = strings.Replace(rest,
+			spreadToken,
+			spanToken, 1)
+	}
+
+	if strings.Contains(rest, spanToken) {
+		if rest == "-" {
+			// whole day span
+			parsed.Start = Clock{0, 0}
+			parsed.End = Clock{24, 0}
+		} else {
+			parsed.Start, parsed.End, err = parseClockRange(rest)
+		}
+	} else {
+		parsed.Start, err = ParseClock(rest)
+		if err == nil {
+			parsed.End = parsed.Start
+		}
+	}
+
+	if err != nil {
+		return span, fmt.Errorf("cannot parse %q: not a valid time", s)
+	}
+
+	return parsed, nil
+}
+
+// parseWeekday parses a single weekday (eg. wed, mon5),
+func parseWeekday(s string) (week Week, err error) {
+	l := len(s)
+	if l != 3 && l != 4 {
+		return week, fmt.Errorf("cannot parse %q: invalid format", s)
+	}
+
+	var day = s
+	var pos uint
+	if l == 4 {
+		day = s[0:3]
+		v, err := strconv.ParseUint(s[3:], 10, 32)
+		if err != nil || v < 1 || v > 5 {
+			return week, fmt.Errorf("cannot parse %q: invalid week number", s)
+		}
+		pos = uint(v)
+	}
+
+	weekday, ok := weekdayMap[day]
+	if !ok {
+		return week, fmt.Errorf("cannot parse %q: invalid weekday", s)
+	}
+
+	return Week{weekday, pos}, nil
+}
+
+// parseCount will parse the string containing a count token and return the
+// count, the rest of the string with count information removed or an error
+func parseCount(s string) (count uint, rest string, err error) {
+	if !strings.Contains(s, countToken) {
+		return 0, s, nil
+	}
+
+	// timespan = time ( "-" / "~" ) time [ "/" ( time / count ) ]
+	split := strings.Split(s, countToken)
+	if len(split) != 2 {
+		return 0, "", fmt.Errorf("cannot parse %q: invalid event count", s)
+	}
+
+	rest = split[0]
+	countStr := split[1]
+	c, err := strconv.ParseUint(countStr, 10, 32)
+	if err != nil || c == 0 {
+		return 0, "", fmt.Errorf("cannot parse %q: invalid event interval", s)
+	}
+	return uint(c), rest, nil
+}
+
+const (
+	spanToken   = "-"
+	spreadToken = "~"
+	countToken  = "/"
+)
+
+var shorthandTime = map[string]bool{
+	// shorthand variant of whole day
+	"-": true,
+	// and whole day with spreading
+	"~": true,
+}
+
+// Parse each event and return a slice of schedules.
+func parseEventSet(s string) (*Schedule, error) {
+	var events []string
+	// split event set into events
+	//     eventset = wdaylist / timelist / wdaylist "," timelist
+	// or wdaysets
+	//     wdaylist = wdayset *( "," wdayset )
+	// or timesets
+	//     timelist = timeset *( "," timeset )
+	//
+	// NOTE: the syntax is ambiguous in the sense the type of a 'set' is now
+	// explicitly indicated, events with : inside are expected to be
```

Comment:
Same here. Please revise the terminology and comments to make sure we're not referring to these pieces as events.